Robot2019/src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -7,78 +7,88 @@
-      WPI_VictorSPX r3, Joystick leftJoy, Joystick rightJoy) {
-    leftMaster = l1;
-    rightMaster = r1;
-    leftSlave1 = l2;
-    leftSlave2 = l3;
-    rightSlave1 = r2;
-    rightSlave2 = r3;
+  public Drivetrain(ArrayList<WPI_TalonSRX> masters, ArrayList<WPI_VictorSPX> slaves, ArrayList<Joystick> joys,
+      ArrayList<Encoder> encs) {
+    leftMaster = masters.get(0);
+    rightMaster = masters.get(1);
+    leftSlave1 = slaves.get(0);
+    leftSlave2 = slaves.get(1);
+    rightSlave1 = slaves.get(2);
+    rightSlave2 = slaves.get(3);
 
-    this.leftJoy = leftJoy;
-    this.rightJoy = rightJoy;
+    leftJoy = joys.get(0);
+    rightJoy = joys.get(1);
+
+    leftEnc = encs.get(0);
+    rightEnc = encs.get(1);
 
-    drive_mode = "arcade";
-    lvalue = 0.0;
-    rvalue = 0.0;
     leftK = 1.0;
     rightK = 1.0;
+
+    drive_mode = "Arcade";
+  }
+
+  public void TeleopDrive(double val1, double val2) {
+      leftMaster.set(Math.signum(val1) * leftK * val1 * val1);
+      rightMaster.set(Math.signum(val2) * rightK * val2 * val2);
   }
 
-  public void TeleopDrive() {
-    lvalue = leftJoy.getY();
-    rvalue = rightJoy.getY();
-    // NOTE: The problem with this code is that it relies on the driver
-    // to not touch the right joystick if we want the robot to go forward
-    if (drive_mode.equals("arcade")) {
-      if (rvalue == 0.0) {          // Forward / Back
-        leftMaster.set(Math.signum(lvalue) * leftK * lvalue * lvalue);
-        rightMaster.set(Math.signum(lvalue) * rightK * lvalue * lvalue);
-      } else if (rvalue > 0.0) {    // Turn Left
-        leftMaster.set(Math.signum(lvalue) * leftK * lvalue * lvalue);
-        rightMaster.set(0.0);
-      } else if (rvalue < 0.0) {    // Turn Right
-        leftMaster.set(0.0);
-        rightMaster.set(Math.signum(lvalue) * rightK * lvalue * lvalue);
-      }
+  public double getEncDist(String type) {
+    if (type.equals("left")) {
+      return leftEnc.getDistance();
     }
-    else if (drive_mode.equals("tank")) {
-        leftMaster.set(Math.signum(lvalue) * leftK * lvalue * lvalue);
-        leftMaster.set(Math.signum(rvalue) * rightK * rvalue * rvalue);
+    else {
+      return rightEnc.getDistance();
+    }
+  }
+
+  public double getEncRate(String type) {
+    if (type.equals("left")) {
+      return leftEnc.getRate();
+    }
+    else {
+      return rightEnc.getRate();
     }
   }
 
   @Override
   public void initDefaultCommand() {
-    setDefaultCommand(new TeleopDrive(this));
+    if (drive_mode.equals("Arcade")) {
+      setDefaultCommand(new ArcadeDrive(this));
+    } else {
+      setDefaultCommand(new TankDrive(this));
+    }
   }
 
-  public void setDriveMode(String drivemode) {
-    drive_mode = drivemode;
+  public void changeDriveMode() {
+    if (drive_mode.equals("Arcade")) {
+      drive_mode = "Tank";
+    } else {
+      drive_mode = "Arcade";
+    }
```

Comment:
There is a chance that this does not update the default command. Can do do more research on this and either a) confirm that the default command updates when you change the string or b) add code that makes sure the default command changes.